nixos/modules/services/networking/kea.nix

```diff
@@ -9,20 +9,26 @@ with lib;
 let
   cfg = config.services.kea;
 
+  xor = x: y: (!x && y) || (x && !y);
   format = pkgs.formats.json {};
 
-  ctrlAgentConfig = format.generate "kea-ctrl-agent.conf" {
+  choose = x: y: if x != null then x else y;
+
+  ctrlAgentConfig = choose cfg.ctrl-agent.configFile (format.generate "kea-ctrl-agent.conf" {
```

Comment:
```suggestion
  ctrlAgentConfig = cfg.ctrl-agent.configFile or (format.generate "kea-ctrl-agent.conf" {
```

Reply:
Not quite. `or` only helps if the attribute was not found, not if it exists but is null.